components/brave_origin/policy/brave_origin_policy_provider.cc

```diff
@@ -0,0 +1,261 @@
+  const base::Value::Dict& policies_dict =
+      local_state_->GetDict(prefs::kBravePolicies);
+  const base::Value* managed_value = policies_dict.Find(externally_managed_key);
+
+  return managed_value && managed_value->is_bool() && managed_value->GetBool();
+}
+
+void BraveOriginPolicyProvider::SetBraveOriginPolicyForPref(
+    policy::PolicyMap& policy_map,
+    const std::string& policy_key,
+    const std::string& pref_name,
+    PrefService* local_state) {
+  // Only set policy if not already set by external providers
+  bool is_external = IsPolicySetByExternalProvider(policy_key);
+
+  if (!is_external) {
+    // Get the brave_policies dictionary and iterate through all keys
+    const base::Value::Dict& policies_dict =
+        local_state->GetDict(brave_origin::prefs::kBravePolicies);
+
+    bool found_policy = false;
+    // Look for keys that end with our pref_name (format: profile_id.pref_name)
+    for (const auto item : policies_dict) {
+      const std::string& dict_key = item.first;
+      const base::Value& policy_value = item.second;
+
+      // Check if this key is for our pref_name
+      // Key format is: profile_id.pref_name
+      // Find first dot (after profile_id) and extract everything after it
+      size_t first_dot = dict_key.find('.');
+      if (first_dot != std::string::npos) {
+        std::string key_pref_name = dict_key.substr(first_dot + 1);
+        if (key_pref_name == pref_name) {
+          policy_map.Set(policy_key, policy::POLICY_LEVEL_MANDATORY,
+                         policy::POLICY_SCOPE_MACHINE,
+                         policy::POLICY_SOURCE_PLATFORM, policy_value.Clone(),
+                         nullptr);
+          found_policy = true;
+          break;  // Use first found value
+        }
+      }
+    }
+
+    if (!found_policy) {
+      LOG(ERROR) << "No policy value found for pref: " << pref_name;
+    }
+  }
+}
+
+void BraveOriginPolicyProvider::SetBraveOriginGlobalPolicyForPref(
+    policy::PolicyMap& policy_map,
+    const std::string& policy_key,
+    const std::string& pref_name,
+    PrefService* local_state) {
+  // Only set policy if not already set by external providers
+  bool is_external = IsPolicySetByExternalProvider(policy_key);
+
+  if (!is_external) {
+    // For global policies, use the BraveOrigin default values from policy
+    // definitions
+    const BraveOriginPrefInfo* pref_info =
+        BraveOriginService::GetPrefInfo(pref_name);
+    if (pref_info) {
+      // Use the default value from the policy definition
+      base::Value policy_value = pref_info->default_value.Clone();
+
+      policy_map.Set(policy_key, policy::POLICY_LEVEL_MANDATORY,
+                     policy::POLICY_SCOPE_MACHINE,
+                     policy::POLICY_SOURCE_PLATFORM, policy_value.Clone(),
+                     nullptr);
+
+      // ALSO manually apply the policy to local_state pref since policy
+      // handlers won't process profile-level policies for global prefs
```

Comment:
Wouldn't global prefs be controlled by policies that are not "profile-level"? i.e. the `BraveStatsPingEnabled` is not per profile

From my POV, it seems that the policy service would set the local state pref value without any issue, but perhaps I'm missing something.

Reply:
Actually I think we need to keep these separate because otherwise we'd need to store defaults outside of registry. We can but that would probably introduce a map for that